### PR TITLE
Method for retrieving available optimizer rules

### PR DESCRIPTION
**Pull request by cpjulia**
### Scope & Purpose

Obs.: more to be added in the next commits 
This PR introduces a GET method for returning the available optimizer rules and signalling whether they are only made for enterprise or not with flag `EnterpriseOnly`. 

- [ ] :hankey: Bugfix
- [x] :pizza: New feature
- [ ] :fire: Performance improvement
- [ ] :hammer: Refactoring/simplification

### Checklist

- [x] Tests
  - [ ] **Regression tests**
  - [ ] C++ **Unit tests**
  - [x] **integration tests**
  - [ ] **resilience tests**
- [ ] :book: CHANGELOG entry made
- [ ] :books: documentation written (release notes, API changes, ...)
- [ ] Backports
  No backports required.

[x] Docs PR: https://github.com/arangodb/docs/pull/947

### CODE REVIEW

File: arangod/RestHandler/RestQueryHandler.cpp [hunk 1]
```diff
@@ -73,6 +78,35 @@ RestStatus RestQueryHandler::execute() {
   return RestStatus::DONE;
 }
 
+void RestQueryHandler::handleAvailableOptimizerRules() {
+  VPackBuilder builder;
+  builder.openArray();
+  auto& rules = OptimizerRulesFeature::rules();
```
Comment (jsteemann) [inline arangod/RestHandler/RestQueryHandler.cpp:84]:
```suggestion
  auto const& rules = OptimizerRulesFeature::rules();
```

File: tests/js/client/shell/shell-optimizer-rules.js [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+// / WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// / See the License for the specific language governing permissions and
+// / limitations under the License.
+// /
+// / Copyright holder is triAGENS GmbH, Cologne, Germany
+// /
+// / @author Julia Puget
+// //////////////////////////////////////////////////////////////////////////////
+
+let jsunity = require('jsunity');
```
Comment (jsteemann) [inline tests/js/client/shell/shell-optimizer-rules.js:28]:
```suggestion
const jsunity = require('jsunity');
```

File: tests/js/client/shell/shell-optimizer-rules.js [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+
+function optimizerRulesSuite() {
+  'use strict';
+  return {
+
+    testDisplayingOfAllRules: function () {
+      let response = arango.GET("/_api/query/rules");
+      let clusterOnlyRules = expectedRules.filter(rule => rule.flags.clusterOnly === true);
+      let nonClusterOnlyRules = expectedRules.filter(rule => rule.flags.clusterOnly === false);
+      nonClusterOnlyRules.forEach(nonClusterRule => {
```
Comment (jsteemann) [inline tests/js/client/shell/shell-optimizer-rules.js:110]: Can we also add a check here that `clusterOnlyRules.length > 0` && `nonClusterOnlyRules.length > 0`.

Comment (cpjulia) [reply tests/js/client/shell/shell-optimizer-rules.js:110]: Addressed

File: arangod/RestHandler/RestQueryHandler.cpp [hunk 4]
```diff
@@ -73,6 +78,35 @@ RestStatus RestQueryHandler::execute() {
...
+                VPackValue(optimizerRule.isClusterOnly() ? true : false));
+    builder.add("canBeDisabled",
+                VPackValue(optimizerRule.canBeDisabled() ? true : false));
+    builder.add(
+        "canCreateAdditionalPlans",
+        VPackValue(optimizerRule.canCreateAdditionalPlans() ? true : false));
+    builder.add("disabledByDefault",
+                VPackValue(optimizerRule.isDisabledByDefault() ? true : false));
+    builder.add("enterpriseOnly",
+                VPackValue(optimizerRule.isEnterpriseOnly() ? true : false));
```
Comment (jsteemann) [inline arangod/RestHandler/RestQueryHandler.cpp:101]:
```suggestion
    builder.add("hidden", VPackValue(optimizerRule.isHidden()));
    builder.add("clusterOnly",
                VPackValue(optimizerRule.isClusterOnly()));
    builder.add("canBeDisabled",
                VPackValue(optimizerRule.canBeDisabled()));
    builder.add(
        "canCreateAdditionalPlans",
        VPackValue(optimizerRule.canCreateAdditionalPlans()));
    builder.add("disabledByDefault",
                VPackValue(optimizerRule.isDisabledByDefault()));
    builder.add("enterpriseOnly",
                VPackValue(optimizerRule.isEnterpriseOnly()));
```

**Review by jsteemann — APPROVED** [review]
LGTM!

**Pull request merged**